File: go/hamming/hamming.go
//Package hamming calculates hamming distance
package hamming

import "errors"

//Distance calculates hamming distance
func Distance(a, b string) (int, error) {
	hammingDistance := 0

	if len(a) != len(b) {
		return hammingDistance, errors.New("strings must be the same length")
	}

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			hammingDistance++
		}

	}

	return hammingDistance, nil
}

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    private final String leftStrand;
    private final String rightStrand;

    public Hamming(String leftStrand, String rightStrand) {
        if (leftStrand == null
            || rightStrand == null
            || leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException(
                    "leftStrand and rightStrand must be of equal length.");
        }

        this.leftStrand = leftStrand;
        this.rightStrand = rightStrand;
    }

    int getHammingDistance() {
        int hammingDistance = 0;
        for(int i = 0; i < leftStrand.length(); i++) {
            if(leftStrand.charAt(i) != rightStrand.charAt(i))
                hammingDistance++;
        }
        return hammingDistance;
    }

}
